modules/internal/SharedObject.chpl

```diff
@@ -323,6 +316,41 @@ module SharedObject {
       chpl_pn = nil;
     }
 
+    // Issue a compiler error for illegal uses.
+    pragma "no doc"
+    proc type create(source) {
+      compilerError("cannot create a 'shared' from ", source.type:string);
+    }
+
+    /* Changes the memory management strategy of the argument from `owned`
+       to `shared`, taking over the ownership of the argument.
+       The result type preserves nilability of the argument type.
+       If the argument is non-nilable, it must be recognized by the compiler
+       as an expiring value. */
+    inline proc type create(pragma "nil from arg" in take: owned) {
+      var result : shared = take;
+      return result;
+    }
+
+    /* Creates a new `shared` class reference to the argument.
+       The result has the same type as the argument. */
+    inline proc type create(pragma "nil from arg" const ref src: shared) {
```

Comment:
This should use `in` intent on the argument. Doing so allows the compiler to minimize the reference counts.